apps/system/test/unit/mock_app.js

```diff
@@ -70,7 +70,13 @@ MockApp.prototype.mTriggerDownloadError = function(error) {
     });
   }
 };
+MockApp.prototype.addEventListener = function(evt, callback) {
+callback();
```

Comment:
in the future it might be better to fire events from the test rather than to just blindly execute `callback` when `addEventListener` is called